### PR TITLE
Add downcase, upcase, and number-formatter missing symbols

### PR DESCRIPTION
**Pull request by sofita**
A style using `downcase` filter expression was erroring due to a missing symbol `lowercase`.

This PR adds the function implementation for `lowercase` as well as that for `upcase` and `number-formatter` which were also missing in the binary. 

The gl-native vendored nunicode library was added to support the string functions that require nunicode functions.

### CODE REVIEW

**Comment by sofita** [comment]
## Running tests on master branch to show that they fail

First test:
image

Second test:
image

Third test (number-format gets hit first, because it is the first one in the style):
image

**Comment by sofita** [comment]
## Testing on staging

Style that filters on symbolrank being 7, and downcase name including `san`
image
image

Request succeeds
image


## Reverting staging to not use latest vtshaver
Cannot reproduce, requests are coming back as 200 😕

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-16
```
Comment (johnk1) [inline .nvmrc:1]: I didn't mean to add this to the branch. Can you revert just to 16?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## v0.3.2
+- Add missing symbol implementations for `downcase`, `upcase`, and `number-formatter` expressions [#58](https://github.com/mapbox/vtshaver/pull/58)
```
Comment (johnk1) [inline CHANGELOG.md:4]: It might be worth mentioning that `number-formatter` is very basic and doesn't use any of the config options passed in.

**Review by johnk1 — CHANGES_REQUESTED** [review]
Just 2 things: we should stick to .nvmrc = 16 and I have a suggest re: changelog.

**Review by johnk1 — APPROVED** [review]
🎉 thanks!

**Comment by sofita** [comment]
> @sofita @johnk1 Instead of adding all the nunicode headers, can we just add mason dependency for nunicode-1.8.0? I believe mason-versions.ini file is there just for that. I don't remember the nitty-gritties there but I think you can just add `nunicode=1.8` in mason-versions.ini and mention the downloaded headers directly in the system_includes just like mbgl headers.
> 
> Didn't think this through but I reckon that would be a better approach especially when the mason package is prebuilt. What do you think?

I actually tried this exact thing! `nunicode=1.8` in mason-versions but it didn't seem to work for me

**Comment by axrj** [comment]
> > @sofita @johnk1 Instead of adding all the nunicode headers, can we just add mason dependency for nunicode-1.8.0? I believe mason-versions.ini file is there just for that. I don't remember the nitty-gritties there but I think you can just add `nunicode=1.8` in mason-versions.ini and mention the downloaded headers directly in the system_includes just like mbgl headers.
> > Didn't think this through but I reckon that would be a better approach especially when the mason package is prebuilt. What do you think?
> 
> @axrj I actually tried this exact thing! `nunicode=1.8` in mason-versions but it didn't seem to work for me


Ah right. Haven't checked on the history of this PR. did you try adding the include dir in the [system_includes](https://github.com/mapbox/vtshaver/blob/071c89aa4b2994f60ee7b70a5bbe7a4a39db8640/binding.gyp#L17) ? 
what error did you see? is it that the runtime symbols could not be found or build didn't go through?

**Comment by sofita** [comment]
> Ah right. Haven't checked on the history of this PR. did you try adding the include dir in the [system_includes](https://github.com/mapbox/vtshaver/blob/071c89aa4b2994f60ee7b70a5bbe7a4a39db8640/binding.gyp#L17) ? what error did you see? is it that the runtime symbols could not be found or build didn't go through?
> 
> I won't hold this PR regardless as this needs to be fixed soon in production. I'll try to see later if it can be added as a mason dependency instead.

I did not add to system_includes . This line (include dir) is already there `"-isystem <(module_root_dir)/mason_packages/.link/include",` was there another line to add? I don't know what it would be for nunicode

**Comment by sofita** [comment]
> > Ah right. Haven't checked on the history of this PR. did you try adding the include dir in the [system_includes](https://github.com/mapbox/vtshaver/blob/071c89aa4b2994f60ee7b70a5bbe7a4a39db8640/binding.gyp#L17) ? what error did you see? is it that the runtime symbols could not be found or build didn't go through?
> > I won't hold this PR regardless as this needs to be fixed soon in production. I'll try to see later if it can be added as a mason dependency instead.
> 
> I did not add to system_includes . This line (include dir) is already there `"-isystem <(module_root_dir)/mason_packages/.link/include",` was there another line to add? I don't know what it would be for nunicode

Will make another ticket for this, so that we can get shaving back on master

**Pull request merged**